### PR TITLE
Fix GC test failures

### PR DESCRIPTION
**Pull request by sywhang**
This addresses GC test failures introduced from #17680.

### CODE REVIEW

**Comment by BruceForstall** [comment]
@dotnet-bot test Windows_NT x64 Build and Test
@dotnet-bot test Windows_NT x86 Checked Build and Test

File: tests/src/GC/API/GCHandle/HandleCopy.cs [hunk 1]
```diff
@@ -38,17 +37,13 @@ public CreateObj()
...
-            obj = null;
-        }    
-
         public bool RunTest()
         {
+            // ensuring that GC happens even with /debug mode
+            obj = null;
             GC.Collect();
+
             GC.WaitForPendingFinalizers();
```
Comment (Maoni0) [inline tests/src/GC/API/GCHandle/HandleCopy.cs]: I don't understand the comment... we want to ensure GC happens with /debug mode by reverting the code?

Comment (sywhang) [reply tests/src/GC/API/GCHandle/HandleCopy.cs]: Oh I think I made an outdated commit here actually... I just reverted this test to what it was before to see the change in behavior but I didn't add the changes I made to this branch to eventually fix it. Thanks for spotting that!

**Comment by sywhang** [comment]
@dotnet-bot test Windows_NT x64 Build and Test
@dotnet-bot test Windows_NT x86 Checked Build and Test

**Comment by erozenfeld** [comment]
@Maoni0 Can you please review this? If it looks good, I'd like to get this merged to make lots of pri 1 jobs greener.

**Comment by sywhang** [comment]
@dotnet-bot test Windows_NT x64 Build and Test
@dotnet-bot test Windows_NT x86 Checked Build and Test

**Comment by sywhang** [comment]
@dotnet-bot test Windows_NT x64 Build and Test
@dotnet-bot test Windows_NT x86 Checked Build and Test

**Comment by erozenfeld** [comment]
@dotnet-bot test OSX10.12 x64 Checked Innerloop Build and Test

**Pull request merged**